fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1060,7 +1060,10 @@ private void handleKillQuery(String queryId) throws DdlException {
             context.getState().setError(errorMsg == null ? ErrorCode.ERR_UNKNOWN_ERROR.formatErrorMsg() : errorMsg);
             return;
         }
-        ConnectContext killCtx = ExecuteEnv.getInstance().getScheduler().findContextByQueryId(queryId);
+        ConnectContext killCtx = ExecuteEnv.getInstance().getScheduler().findContextByCustomQueryId(queryId);
+        if (killCtx == null) {
+            killCtx = ExecuteEnv.getInstance().getScheduler().findContextByQueryId(queryId);
+        }
```

Comment:
Because this feature is built for expert user who is a platform user, the user will set `custom_query_id` and kill query by this id. Killing query by query id that assigned by starrocks is less happening comparing to killing query by `custom_query_id`

Reply:
Got it.
The custom_query_id default value is `""`, we need to avoid killing the wrong queries that `kill query ""`.
We need to add a protect logic that we do nothing when the query_id equals to the custom_query_id default value.

Reply:
Done.